server/controllers/api.py

```diff
@@ -1026,6 +1050,54 @@ class File(Resource):
     def get(self, user, file_id):
         return files.file_download(file_id, user)
 
+class Client(Resource):
+    """ View OAuth clients programmatically.
+        Authenticated. Permissions: >= Staff owning the client
+    """
+    schema = ClientSchema()
+    model = models.Client
+
+    @marshal_with(schema.client_fields)
+    def get(self, user, client_id):
+        client = models.Client.query.get(client_id)
+        if not client:
+            if user.is_admin:
+                restful.abort(404)
+            else:
+                restful.abort(403)
+        if not models.Client.can(client, user, "view"):
+            restful.abort(403)
+        return {
+            'client_name': client.name,
+            'owner_email': client.user.email,
+            'description': client.description,
+            'client_id': client.client_id,
+            'is_confidential': client.is_confidential,
+            'allowed_redirects': client.redirect_uris,
+        }
```

Comment:
wait did this not how exist? how does the current UI element work?

Reply:
This is just for the API. `models.Client` always existed, and was used directly in the UI.